# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/README.md
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/README.md b/third_party/rust/naga/README.md
--- a/third_party/rust/naga/README.md
+++ b/third_party/rust/naga/README.md
@@ -19,17 +19,17 @@ SPIR-V (binary) | :white_check_mark: | s
 WGSL            | :white_check_mark: | wgsl-in | Fully validated |
 GLSL            | :ok:               | glsl-in | |
 
 Back-end        |       Status       | Feature  | Notes |
 --------------- | ------------------ | -------- | ----- |
 SPIR-V          | :white_check_mark: | spv-out  |       |
 WGSL            | :ok:               | wgsl-out |       |
 Metal           | :white_check_mark: | msl-out  |       |
-HLSL            | :construction:     | hlsl-out | Shader Model 5.0+ (DirectX 11+) |
+HLSL            | :ok:               | hlsl-out | Shader Model 5.0+ (DirectX 11+) |
 GLSL            | :ok:               | glsl-out |       |
 AIR             |                    |          |       |
 DXIL/DXIR       |                    |          |       |
 DXBC            |                    |          |       |
 DOT (GraphViz)  | :ok:               | dot-out  | Not a shading language |
 
 :white_check_mark: = Primary support — :ok: = Secondary support — :construction: = Unsupported, but support in progress
 
@@ -61,8 +61,20 @@ are indeed valid for the target platform
 ```bash
 make validate-spv # for Vulkan shaders, requires SPIRV-Tools installed
 make validate-msl # for Metal shaders, requires XCode command-line tools installed
 make validate-glsl # for OpenGL shaders, requires GLSLang installed
 make validate-dot # for dot files, requires GraphViz installed
 make validate-wgsl # for WGSL shaders
 make validate-hlsl # for HLSL shaders. Note: this Make target makes use of the "sh" shell. This is not the default shell in Windows.
 ```
+
+## MSRV
+
+The `naga` codebase's MSRV is 1.43. However some newer versions of our dependencies have newer MSRVs than that. Here are a list of all known MSRV breaking dependencies and the versions that hold to MSRV.
+
+- `bitflags`: `>1.3` have an MSRV of 1.46. `<=1.2` has an MSRV of 1.43 or earlier.
+
+If you want to use `naga` with `1.43` add the following to your Cargo.toml dependency list even if you don't use bitflags yourself:
+
+```toml
+bitflags = "<1.3"
+```